### PR TITLE
release 2.4.10

### PR DESCRIPTION
**Pull request by donnyyung**

### CODE REVIEW

File: docs/pre-release/reference/intercepts/index.md [hunk 1]
```diff
@@ -68,14 +68,25 @@ your cluster; if it detects it correctly, may simply press "enter" and
...
-created, the traffic-agent will use the result it got from checking
-the in-cluster service.
+When creating an intercept with the `http` mechanism, Telepresence must
+determine the application protocol to use (HTTP/1.1 or HTTP/2). If the
+service's `ports.appProtocol` field is set, it will be used. If not, then
+Telepresence will use the configured [application protocol strategy](../config/#intercept)
+to determine the protocol. The default behavior (`http2Probe` strategy) is
+to send a `GET /telepresence-http2-check` request to your service in order
+to determine if it supports HTTP/2.  This is required for the intercepts to
+behave correctly.
```
Comment (JWKartheiser) [inline docs/pre-release/reference/intercepts/index.md:78]: Just some minor tweaks:
_When you create an intercept with the http mechanism, Telepresence  determines whether the application protocol uses HTTP/1.1 or HTTP/2. If the service's ports.appProtocol field is set, Telepresence uses that. If not, then Telepresence uses the configured application protocol strategy to determine the protocol. The default behavior (http2Probe strategy) sends a GET /telepresence-http2-check request to your service to determine if it supports HTTP/2. This is required for the intercepts to behave correctly._

File: docs/pre-release/reference/intercepts/index.md [hunk 2]
```diff
@@ -68,14 +68,25 @@ your cluster; if it detects it correctly, may simply press "enter" and
...
+
+### TLS
+
+If the intercepted service expects TLS, then Telepresence must terminate the
+TLS in order to use the `http` mechanism. It is then common to also restore
+the TLS by originating it again. This is controlled using [TLS annotations](../cluster-config/#tls)
+in the workload.
```
Comment (JWKartheiser) [inline docs/pre-release/reference/intercepts/index.md:85]: Just trimmed down the verbiage a bit:
_If the intercepted service expects TLS, then Telepresence terminates the TLS to use the http mechanism. It is then common to also restore the TLS by originating it again. This is controlled using TLS annotations in the workload._

Quick question: "It is then common to also restore the TLS by originating it again" seems like a potentially confusing statement. Is that done automatically? If so, it's worth mentioning.

File: docs/v2.4/reference/config.md [hunk 3]
```diff
@@ -134,6 +135,29 @@ The `telepresenceAPI` controls the behavior of Telepresence's RESTful API server
...
+
+| Protocol | Meaning                               |
+|----------|---------------------------------------|
+| `http`   | Plaintext HTTP/1.1 traffic            |
+| `http2`  | Plaintext HTTP/2 traffic              |
+| `https`  | TLS Encrypted HTTP (1.1 or 2) traffic |
+| `grpc`   | Same as http2                         |
+
 ## Per-Cluster Configuration
 Some configuration is not global to Telepresence and is actually specific to a cluster.  Thus, we store that config information in your kubeconfig file, so that it is easier to maintain per-cluster configuration.
```
Comment (JWKartheiser) [inline docs/v2.4/reference/config.md:163]: [Same comments as in the pre-release version of the file.]

File: docs/v2.4/reference/intercepts/index.md [hunk 4]
```diff
@@ -68,14 +68,25 @@ your cluster; if it detects it correctly, may simply press "enter" and
...
+If the intercepted service expects TLS, then Telepresence must terminate the
+TLS in order to use the `http` mechanism. It is then common to also restore
+the TLS by originating it again. This is controlled using [TLS annotations](../cluster-config/#tls)
+in the workload.
+
+Use the `--http-plaintext` flag when doing an intercept when the service in the
+cluster is using TLS in case you want to use plaintext for the communication with the
+process on your local workstation. 
 
 ## Supported workloads
```
Comment (JWKartheiser) [inline docs/v2.4/reference/intercepts/index.md:92]: [Same comments as in the pre-release version of the file.]

**Review by JWKartheiser — COMMENTED** [review]
Looks really solid to me. A handful of suggestions for slight changes to some of the sentences, but otherwise it looks like it's ready to roll. 👍

File: docs/pre-release/reference/config.md [hunk 5]
```diff
@@ -134,6 +135,29 @@ The `telepresenceAPI` controls the behavior of Telepresence's RESTful API server
 If the `traffic-manager` is auto-installed, its webhook agent injector will be configured to add the `TELEPRESENCE_API_PORT` environment to the app container when the `traffic-agent` is injected.
 See [RESTful API server](../restapi) for more info.
 
+#### Intercept
+The `intercept` controls applies to how telepresence will intercept the communications to the intercepted service.
+
+The `defaultPort` controls what port that will be selected when no `--port` flag is given to the `telepresence intercept` command. The default value is "8080".
```
[the remark this answers is not in the document]
Comment (donnyyung) [reply docs/pre-release/reference/config.md:141]: So I think I'm still a little confused with the revision, the defaultPort is only used when a user does an intercept without the `--port` flag. 

So for example if defaultPort:9090

```
telepresence intercept --port 1010 <service>
```
will create an intercept on port 1010
```
telepresence intercept <service>
```
would create an intercept on port 9090.. does that make sense?

File: docs/pre-release/reference/intercepts/index.md (hunk 2 above)
Comment (donnyyung) [reply docs/pre-release/reference/intercepts/index.md:85]: Yeah it's not done automatically, which is why the original linked to the TLS annotations section bc users need to follow the instructions there (e.g. adding annotations) to their workload. So should we add that back in or make it more clear that they actually need to go to that page and do what the docs there suggest?

**Review by JWKartheiser — APPROVED** [review]
Love it. Approved.

File: docs/pre-release/reference/intercepts/index.md (hunk 2 above)
Comment (donnyyung) [reply docs/pre-release/reference/intercepts/index.md]: oh whoops I hadn't removed the first version of this from the file. So thanks for calling my attention to this. that has been fixed!

**Pull request merged**